src/App.tsx

```diff
@@ -21,9 +38,22 @@ const App = () => {
   const [drawerOpen, setDrawerOpen] = useState(isGreaterThanLgBreakpoint());
 
   useEffect(() => {
-    const isAccessTokenPresent = getAuthSession('accessToken') !== null;
-    const isTokenExpired = checkAuthStatusService();
-    setAuthenticated(isTokenExpired || isAccessTokenPresent);
+    const checkAuth = async () => {
+      const isAccessTokenPresent = getAuthSession('accessToken') !== null;
+      let isSessionAlive = false;
+      if(!isAccessTokenPresent){
+        // New user
+        isSessionAlive = false;
+      }
+      else{
+        // existing user
+        isSessionAlive = await checkSessionValidity();
+      }
+      startTransition(() => {
```

Comment:
Do we need startTransition? I don't see how this will be helpful in our context.

Reply:
The tests were raising warnings and recommending the use of StartTransition.